soh/src/code/z_actor.c

```diff
@@ -3142,6 +3143,40 @@ int gMapLoading = 0;
 
 Actor* Actor_Spawn(ActorContext* actorCtx, GlobalContext* globalCtx, s16 actorId, f32 posX, f32 posY, f32 posZ,
                    s16 rotX, s16 rotY, s16 rotZ, s16 params) {
+
+    // Hack to remove bats and skulltulas that spawn in graveyard because of removing object dependency.
+    if ((actorId == ACTOR_EN_FIREFLY || (actorId == ACTOR_EN_SW && params == 0)) && globalCtx->sceneNum == SCENE_SPOT02) {
+        return NULL;
+    }
+
+    uint8_t tryRandomizeEnemy = CVar_GetS32("gRandomizedEnemies", 0) && gSaveContext.fileNum >= 0 && gSaveContext.fileNum <= 2;
+
+    if (tryRandomizeEnemy) {
+        if (IsEnemyFoundToRandomize(actorId, params)) {
+
+            // Big jellyfish spawn too high up, this fixes that.
+            if (actorId == ACTOR_EN_VALI) {
+                posY = -330;
+            }
+
+            enemyEntry newEnemy = GetRandomizedEnemy();
```

Comment:
> Replacing guays, bats, skulltulas or other enemies that normally spawn in the sky don't work very well with any enemies that don't obey gravity. A beamos floating in the sky isn't going to do a whole lot. Unsure still what the best way would be to handle this.

We already know which enemies don't obey gravity, right?  If it were me, I'd wrap this Get call in a loop, and if the old actor is one that spawned in the air and the new one is one that doesn't obey gravity, then just retry the Get call until we get a new one that obeys gravity.  In theory, such a loop could run forever, but in practice I think it would be fine since (I assume) most enemies obey gravity.

Reply:
There's more enemies that don't obey gravity than you might think at first, so I'm not a huge fan of limiting what actors to spawn where. I think a better solution would be to either somehow place enemies down on the ground after their init function when enemy randomizer is on. Which could admittedly be quite a bit of work because I'd have to add it seperately to each relevant enemy. 

Other option would be to somehow do a raycast inside of Actor_Spawn() and place actors down on the ground that way. I don't actually know if the second method could work because it's spawning the actors while the map is still loading in though.

So yeah, not sure what the best way to go about this would be for now, I might have to experiment a bit with it. Open for new ideas though.

Reply:
Solved this by doing a raycast to place randomized enemies on the ground.